gating/check/pre_deploy.sh

```diff
@@ -60,8 +60,21 @@ if [[ ! -e phantomjs-2.1.1-linux-x86_64.tar.bz2 ]]; then
   tar -xjf phantomjs-2.1.1-linux-x86_64.tar.bz2
 fi
 
+# FrankZhang: Limit venv python to python2 
+# to avoid python3 code making syntax error https://rpc-openstack.atlassian.net/browse/RI-458
 if [[ ! -d ".venv" ]]; then
-  virtualenv .venv
+  virtualenv .venv --python=python2 --always-copy --no-download
+fi
+
+# FrankZhang: Upgrade pip and setuptools due to outdate issue https://rpc-openstack.atlassian.net/browse/RI-458
+pip_version=$(pip -V | awk '{print $2}')
+if [[ $pip_version < 9.0.1 ]]; then
```

Comment:
the version comparison doesn't work with semver I think.  Though a sting comparison may still work, oddly.

https://stackoverflow.com/a/4024263 may have a better solution
https://stackoverflow.com/a/46299983 is nice too